source/Module/STIG/Functions.Checklist.ps1

```diff
@@ -166,219 +255,227 @@ function New-StigCheckList
 
-    $writer.WriteEndElement(<#STIG_INFO#>)
+        $xccdfBenchmarkContent = Get-StigXccdfBenchmarkContent -Path $XccdfPathItem
 
-    #endregion STIGS/iSTIG/STIG_INFO
+        $stigInfoElements = [ordered] @{
+            'version'        = $xccdfBenchmarkContent.version
+            'classification' = 'UNCLASSIFIED'
+            'customname'     = ''
+            'stigid'         = $xccdfBenchmarkContent.id
+            'description'    = $xccdfBenchmarkContent.description
+            'filename'       = Split-Path -Path $XccdfPathItem -Leaf
+            'releaseinfo'    = $xccdfBenchmarkContent.'plain-text'.InnerText
+            'title'          = $xccdfBenchmarkContent.title
+            'uuid'           = (New-Guid).Guid
+            'notice'         = $xccdfBenchmarkContent.notice.InnerText
+            'source'         = $xccdfBenchmarkContent.reference.source
+        }
 
-    #region STIGS/iSTIG/VULN[]
+        foreach ($stigInfoElement in $stigInfoElements.GetEnumerator())
+        {
+            $writer.WriteStartElement("SI_DATA")
+            $writer.WriteStartElement('SID_NAME')
+            $writer.WriteString($stigInfoElement.name)
+            $writer.WriteEndElement(<#SID_NAME#>)
+            $writer.WriteStartElement('SID_DATA')
+            $writer.WriteString($stigInfoElement.value)
+            $writer.WriteEndElement(<#SID_DATA#>)
+            $writer.WriteEndElement(<#SI_DATA#>)
+        }
 
-    # Pull in the processed XML file to check for duplicate rules for each vulnerability
-    [xml]$xccdfBenchmark = Get-Content -Path $xccdfPath -Encoding UTF8
-    $fileList = Get-PowerStigFileList -StigDetails $xccdfBenchmark
-    $processedFileName = $fileList.Settings.FullName
-    [xml]$processed = Get-Content -Path $processedFileName
+        $writer.WriteEndElement(<#STIG_INFO#>)
 
-    $vulnerabilities = Get-VulnerabilityList -XccdfBenchmark $xccdfBenchmarkContent
+        #endregion STIGS/iSTIG/STIG_INFO
 
-    foreach ($vulnerability in $vulnerabilities)
-    {
-        $writer.WriteStartElement("VULN")
+        #region STIGS/iSTIG/VULN[]
 
-        foreach ($attribute in $vulnerability.GetEnumerator())
-        {
-            $status = $null
-            $findingDetails = $null
-            $comments = $null
-            $manualCheck = $null
+        # Parse out the STIG file name for lookups
+        $stigPathFileName = $XccdfPathItem.Split('\\')
+        $stigFileName = $stigPathFileName[$stigPathFileName.Length-1]
 
-            if ($attribute.Name -eq 'Vuln_Num')
-            {
-                $vid = $attribute.Value
-            }
+        # Pull in the processed XML file to check for duplicate rules for each vulnerability
+        [XML]$xccdfBenchmark = Get-Content -Path $XccdfPathItem -Encoding UTF8
+        $fileList = Get-PowerStigFileList -StigDetails $xccdfBenchmark -Path $XccdfPathItem
+        $processedFileName = $fileList.Settings.FullName
+        [XML]$processed = Get-Content -Path $processedFileName
```

Comment:
a space after the type declaration here.

Reply:
Done.